builtin/providers/aws/resource_aws_sqs_queue.go

```diff
@@ -201,21 +236,35 @@ func resourceAwsSqsQueueRead(d *schema.ResourceData, meta interface{}) error {
 		// iKey = internal struct key, oKey = AWS Attribute Map key
 		for iKey, oKey := range AttributeMap {
 			if attrmap[oKey] != nil {
-				if resource.Schema[iKey].Type == schema.TypeInt {
+				switch resource.Schema[iKey].Type {
+				case schema.TypeInt:
 					value, err := strconv.Atoi(*attrmap[oKey])
 					if err != nil {
 						return err
 					}
 					d.Set(iKey, value)
 					log.Printf("[DEBUG] Reading %s => %s -> %d", iKey, oKey, value)
-				} else {
+				case schema.TypeBool:
+					value, err := strconv.ParseBool(*attrmap[oKey])
+					if err != nil {
+						return err
+					}
+					d.Set(iKey, value)
+					log.Printf("[DEBUG] Reading %s => %s -> %t", iKey, oKey, value)
+				default:
 					log.Printf("[DEBUG] Reading %s => %s -> %s", iKey, oKey, *attrmap[oKey])
 					d.Set(iKey, *attrmap[oKey])
 				}
 			}
 		}
 	}
 
+	// Since AWS does not send the FifoQueue attribute back when the queue
+	// is a standard one (even to false), this enforces the queue to be set
+	// to the correct value.
+	d.Set("fifo_queue", d.Get("fifo_queue").(bool))
```

Comment:
Didn't find any better way to do this without bringing much code complexity. This is a replay in case the attribute is sent back by AWS.